unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Analytics/BIWAnalytics.cs

```diff
@@ -0,0 +1,201 @@
+        events.Add("category Name", catalogItem.categoryName);
+        events.Add("source", source);
+        events.Add("type", catalogItem.itemType.ToString());
+        AddLandInfoSendEvent("new_object_placed", events);
+    }
+
+    public static void QuickAccessAssigned(CatalogItem catalogItem, string source)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("name", catalogItem.name);
+        events.Add("assetPack", catalogItem.assetPackName);
+        events.Add("category", catalogItem.category);
+        events.Add("category Name", catalogItem.categoryName);
+        events.Add("source", source);
+        events.Add("type", catalogItem.itemType.ToString());
+        AddLandInfoSendEvent("quick_access_assigned", events);
+    }
+
+    public static void FavoriteAdded(CatalogItem catalogItem)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("name", catalogItem.name);
+        events.Add("assetPack", catalogItem.assetPackName);
+        events.Add("category", catalogItem.category);
+        events.Add("category Name", catalogItem.categoryName);
+        events.Add("type", catalogItem.itemType.ToString());
+        AddLandInfoSendEvent("favorite_added", events);
+    }
+
+    public static void CatalogItemSearched(string searchQuery, int resultAmount)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("search_query", searchQuery);
+        events.Add("result_amount", resultAmount.ToString());
+        AddLandInfoSendEvent("catalog_item_searched", events);
+    }
+
+    private static Dictionary<string, string> ConvertSceneMetricsModelToDictionary(SceneMetricsModel sceneLimits)
+    {
+        Dictionary<string, string> sceneLimitsDictionary = new Dictionary<string, string>();
+        sceneLimitsDictionary.Add("meshes", sceneLimits.meshes.ToString());
+        sceneLimitsDictionary.Add("bodies", sceneLimits.bodies.ToString());
+        sceneLimitsDictionary.Add("materials", sceneLimits.materials.ToString());
+        sceneLimitsDictionary.Add("textures", sceneLimits.textures.ToString());
+        sceneLimitsDictionary.Add("triangles", sceneLimits.triangles.ToString());
+        sceneLimitsDictionary.Add("entities", sceneLimits.entities.ToString());
+        sceneLimitsDictionary.Add("scene_height", sceneLimits.sceneHeight.ToString());
+
+        return sceneLimitsDictionary;
+    }
+
+    #endregion
+
+    #region CommonInfo
+
+    private static Vector2Int coords;
+    private static string ownership;
+    private static Vector2Int size;
+
+    public static void AddSceneInfo(Vector2Int sceneCoords,  string sceneOwnership, Vector2Int sceneSize)
+    {
+        coords = sceneCoords;
+        ownership = sceneOwnership;
+        size = sceneSize;
+    }
+
+    #endregion
+
+    private static void AddLandInfoSendEvent(string eventName, Dictionary<string, string> events)
```

Comment:
It might add a bit of boilerplate code, but I would suggest only adding data to the dictionary in this method.

Reply:
I don't understand what you mean here, Each function has his own parameters and this function adds the generic one for every event

Reply:
`AddLandInfoSendEvent` -> `AddLandInfo`. So we dont have a method that does two things (adding info to the dictionary and sending the event).

Reply:
This is a common pattern in analytics. The reason behind this decission is to left the ability to add more layers without having to know much about the structure, in a deeper layer more generic information is added like timestamp, user id...etc. 

For example: In this case, I don't need to know about what are the data that we are collecting from the user, I just can wrap my data into the event and send to the next layer 

The name is probably confusing and I will rename it to `SendEditorEvent `  because we want to add this information to every editor event. 

Does this work for you?

Reply:
Sure, but then why dont we just use the `SendEvent` method to add this common data?

Reply:
When we are sending the data from the project panel, we still don't have a land chosen to edit. So we only add the information of the land when we have already chosen a land ( when we enter in editor mode).

We could have 3 methods:
-SendEditorEvent
-SendPanelEvent
-SendEvent

Since the panel don't have common data to share the SendPanelEvent is not added

Reply:
That seems clear enough